retrieverdash/dashboard_script/dashboard_script.py

```diff
@@ -20,20 +20,21 @@
                     'butterfly-population-network',
                     'partners-in-flight',
                     'portal']
-try:
-    dataset_detail = json.load(open('dataset_details.json', 'r'))
-except IOError:
-    with open("dataset_details.json", 'w') as json_file:
-        dataset_detail = dict()
-        json.dump(dataset_detail, json_file)
 
 
+dataset_detail=None
 def check_dataset(dataset):
     os.chdir(os.path.join(file_location))
     md5 = None
     status = None
     reason = None
     try:
+        try:
+            dataset_detail = json.load(open('dataset_details.json', 'r'))
+        except FileNotFoundError:
+            with open("dataset_details.json", 'w') as json_file:
+                dataset_detail = dict()
+                json.dump(dataset_detail, json_file)
```

Comment:
I think we do not need this check here.
```
+            dataset_detail = json.load(open('dataset_details.json', 'r'))
+        except FileNotFoundError:
+            with open("dataset_details.json", 'w') as json_file:
+                dataset_detail = dict()
+                json.dump(dataset_detail, json_file)
```
We have both python and json datasets.
I guess using the dataset objects will give all the objects.

Reply:
Here we are making sure that we have a `dataset_details.json` file in our directory.